CONTRIBUTING.md

```diff
@@ -0,0 +1,328 @@
+# Contributing to DocArray
+
+Thanks for your interest in contributing to DocArray. We're grateful for your initiative! ❤️
+
+In this guide, we're going to go through the steps for each kind of contribution, and good and bad examples of what to do. We look forward to your contributions!
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [🐞 Bugs and Issues](#-bugs-and-issues)
+- [🥇 Making Your First Submission](#-making-your-first-submission)
+- [📝 Code style conventions](#-code-style-conventions)
+- [☑️ Naming Conventions](#-naming-conventions)
+- [💥 Testing DocArray Locally and on CI](#-testing-docarray-locally-and-on-ci)
+- [📖 Contributing Documentation](#-contributing-documentation)
+- [🙏 Thank You](#-thank-you)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+<a name="-bugs-and-issues"></a>
+## 🐞 Bugs and Issues
```

Comment:
Can we change all headings (and corresponding bullets in contents to sentence case? That matches our docs guidelines